site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -15201,7 +15202,19 @@ public void doUnjoin(RunData data) {
 		
 		// refresh the whole page
 		scheduleTopRefresh();
-		
+		String redirectUrl= null;
+		org.sakaiproject.component.api.ServerConfigurationService scs = (org.sakaiproject.component.api.ServerConfigurationService) ComponentManager.get(
+				org.sakaiproject.component.api.ServerConfigurationService.class);
+		//Check if user has access to the site even when they unjoin it, could be beacuse site is public or because of their role.
+		if(site != null && SiteService.allowAccessSite(site.getId())){
+			redirectUrl = getDefaultSiteUrl(ToolManager.getCurrentPlacement().getContext()) + "/" +
```

Comment:
Created a separate method with the name 'getCurrentSiteUrl' instead of using duplicate , pushed the changes.